src/proc.c

```diff
@@ -231,6 +240,24 @@ int get_threads(int pid, int **tids)
     return n;
 }
 
+char *get_thread_states(const int *tids, int n)
+{
+    int i;
+    char *res = calloc(1, n);
+
+    for (i = 0; i < n; ++i) {
+        int state = proc_state(tids[i]);
+        if (state < 0) {
+            fprintf(stderr, "warning: could not get state of thread %d\n",
+                    tids[i]);
+            break;
```

Comment:
Maybe we should continue here?

Reply:
Why not. I changed to set state to `?` in this case. And also squashed this commit with readme update -- as we discussed today, it's really pointless to split it in such a simple case with only single commit.